File: packages/agent_security/android/src/main/kotlin/com/chapter2/agentkit/models/WorldIdProofPayload.kt
package com.chapter2.agentkit.models

public data class WorldIdProofPayload(
    val protocolVersion: String = "4.0",
    val merkleRoot: String,
    val nullifierHash: String,
    val proof: String,
    val credentialType: WorldIdCredentialType = WorldIdCredentialType.Credential11Selfie,
    val action: String,
    val signal: String
)

File: native_security/agentkit/ios/Sources/AgentKitSecurity/Models/WorldIdProofPayload.swift
import Foundation

public struct WorldIdProofPayload: Equatable, Codable, Sendable {
    public let protocolVersion: String
    public let merkleRoot: String
    public let nullifierHash: String
    public let proof: String
    public let credentialType: WorldIdCredentialType
    public let action: String
    public let signal: String

    public init(
        protocolVersion: String = "4.0",
        merkleRoot: String,
        nullifierHash: String,
        proof: String,
        credentialType: WorldIdCredentialType = .credential11Selfie,
        action: String,
        signal: String
    ) {
        self.protocolVersion = protocolVersion
        self.merkleRoot = merkleRoot
        self.nullifierHash = nullifierHash
        self.proof = proof
        self.credentialType = credentialType
        self.action = action
        self.signal = signal
    }
}
